### PR TITLE
CMakeLists.txt: add compile options

### PR DESCRIPTION
**Pull request by Junbo-Zheng**
CMakeLists.txt: add compile options
    
Signed-off-by: Junbo Zheng <3273070@qq.com>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,6 +6,8 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/run)
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/arc)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
 
+add_compile_options(-Wall -Wextra -Werror -Wno-format -g)
```
Comment (Junbo-Zheng) [inline CMakeLists.txt]: Let's mark this PR as draft until all build warnings are fixed.

Comment (buyuer) [reply CMakeLists.txt]: debug模式cmake应该默认加-g

Comment (Junbo-Zheng) [reply CMakeLists.txt]: > debug模式cmake应该默认加-g

Right, I will update the PR later.

Comment (buyuer) [reply CMakeLists.txt]: 也需要区分编译器，因为这几个flag只有gcc和clang支持

Comment (Junbo-Zheng) [reply CMakeLists.txt:9]: > 也需要区分编译器，因为这几个flag只有gcc和clang支持

OK, I will add a check.

Comment (Junbo-Zheng) [reply CMakeLists.txt]: > 也需要区分编译器，因为这几个flag只有gcc和clang支持

OK, follow https://github.com/buyuer/note/blob/main/CMakeLists.txt#L12-L17

Comment (Junbo-Zheng) [reply CMakeLists.txt]: > 也需要区分编译器，因为这几个flag只有gcc和clang支持

Done, I'm not sure if there is a better way to implement it, please review it again

File: CMakeLists.txt [hunk 2]
```diff
@@ -10,9 +10,11 @@ include(cmake/utils.cmake)
 
 if (CMAKE_BUILD_TYPE STREQUAL Debug)
     if (CMAKE_C_COMPILER_ID STREQUAL GCC OR CMAKE_C_COMPILER_ID STREQUAL Clang)
+        add_compile_options(-Wall -Wextra -Werror -Wno-format)
         set(CMAKE_C_FLAGS "-fsanitize=address -fsanitize=leak ${CMAKE_C_FLAGS}")
```
Comment (buyuer) [inline CMakeLists.txt:14]:
```
if (CMAKE_C_COMPILER_ID STREQUAL GCC OR CMAKE_C_COMPILER_ID STREQUAL Clang)
        add_compile_options(-Wall -Wextra -Werror -Wno-format)
endif ()
```

**Pull request merged**